### DISCUSSION TITLE
useQuery enabled not working as expected

### DISCUSSION THREAD

**Discussion by Truis-MF** [TanStack/query#6461] [Q&A]
Hello,

I'm using version 5.10.0 and it appears the the enabled option for useQuery is not working. I'm trying to run a query programatically and utilise the isPending variable to toggle my button. Below is how i'm using the hook...

``` JS
const {refetch, isPending} = useQuery({
    queryKey: ['packingSlip', freightMove.id],
    queryFn: printPackingSlip,
    enabled: false,
});
```

**Comment by TkDodo** [comment]
`isPending` will be `true` but `isLoading` will be false.

**Reply by Truis-MF** [reply]
Thanks for the clarification i'll change that in my code. However the issue regarding useQuery's enabled option is still not working as expected.

**Reply by TkDodo** [reply]
"not working" how? Show a codesandbox reproduction please

**Reply by Truis-MF** [reply]
Looking further into this, the enabled option works are expected when i'm not enabling suspense in the queryClient.

You can see what I'm talking about with this [Codesandbox](https://codesandbox.io/p/devbox/usequery-6461-n87ds7?file=%2Fsrc%2FApp.js%3A15%2C21&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clprglcmk00083b6gvx0gv2ij%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clprglcmk00023b6gt0sj99k2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clprglcmk00053b6g3h6z188v%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clprglcmk00073b6gudjqanzq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clprglcmk00023b6gt0sj99k2%2522%253A%257B%2522id%2522%253A%2522clprglcmk00023b6gt0sj99k2%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clprhpdiq00023b6merrfpggd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A15%252C%2522startColumn%2522%253A21%252C%2522endLineNumber%2522%253A15%252C%2522endColumn%2522%253A21%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clprhpdiq00023b6merrfpggd%2522%257D%252C%2522clprglcmk00073b6gudjqanzq%2522%253A%257B%2522id%2522%253A%2522clprglcmk00073b6gudjqanzq%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clprglcmk00063b6gieapqccs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clprglcmk00063b6gieapqccs%2522%257D%252C%2522clprglcmk00053b6g3h6z188v%2522%253A%257B%2522id%2522%253A%2522clprglcmk00053b6g3h6z188v%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522id%2522%253A%2522clprhmbal02lf3b6gnbeweof5%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clprhmbal02lf3b6gnbeweof5%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Just enable the DevTools network panel and refresh the page, the posts endpoint is being called even when enabled is false.

**Reply by TkDodo** [reply]
the sandbox doesn't start for me, also I wouldn't know what to do with instructions like "Just enable the DevTools network panel" because that means nothing to me.

>  when i'm not enabling suspense in the queryClient.

What I can tell you is that in v5, there is no `enabled` on `useSuspenseQuery`.

**Reply by flowstate** [reply]
This is an unnecessarily snarky reply to a user issue. 

I don't know what to tell you if you're maintaining a request library used primarily in the browser and you don't know what the dev tools panel in a browser is. 

What I can tell you is that this whole conversation comes off sarcastic and unprofessional.

**Reply by frederikhors** [reply]
@flowstate are you joking right? @TkDodo is god! His help was always timely and necessary.

**Reply by TkDodo** [reply]
Thanks for unearthing a discussion from 6 months ago. I don't fully remember what happened here obviously, however, re-reading this discussion:

- the enabled option for useQuery is not working
- useQuery's enabled option is still not working as expected.
- Looking further into this, the enabled option works are expected

All of these statements are not helpful because I don't even know what "expected" is. Apparently it turned out that this was related to suspense, which wasn't mentioned at all.

I get about 100 of such questions per week, and I try to answer them all, in a timely manner, for free. The best way to help me help you is by providing a sandbox that actually works. My only other option is to start ignoring low quality questions, not sure if that would be preferred but it would certainly make my life easier...

**Reply by aurquiel** [reply]
It also doesnt work for me either i mark the useQuery as enabled to false and in the browser tools the query is still is executing i will try to put a sandbox
